src/plugins/data_source/server/saved_objects/data_source_saved_objects_client_wrapper.ts

```diff
@@ -448,9 +448,7 @@ export class DataSourceSavedObjectsClientWrapper {
     auth: T,
     encryptionContext: EncryptionContext
   ) {
-    const {
-      credentials: { username, password },
-    } = auth;
+    const { username, password } = auth as UsernamePasswordTypedContent;
```

Comment:
I am puzzled by this. Does this mean `username` and `password` were props of `auth` and not props of `credentials` which was a prop of `auth`? If so, how was this working until now?

Reply:
Many thanks for pointing this out. It was my mistake and got confused about the structure. Would this work?

`    const {
      credentials: { username, password },
    } = auth as { credentials: UsernamePasswordTypedContent };
`

Reply:
Sorry, but I'm still a bit confused. Even after changing the signature, there's still the below error unless I cast credentials to UsernamePasswordTypedContent.

<img width="1053" alt="Screenshot 2023-12-18 at 10 27 45 PM" src="https://github.com/opensearch-project/OpenSearch-Dashboards/assets/70486227/9c61dd91-f01e-411f-8f1c-d4a127ae6a8a">

Reply:
I checked your latest code, I'm fine with your current solution, the check of `AuthType` should make sure `encryptBasicAuthCredential` to only receive `UsernamePasswordTypedContent` type of credential